scanpy/preprocessing/_pca.py

```diff
@@ -151,19 +157,52 @@ def pca(
 
     logg.info(f'    with n_comps={n_comps}')
 
-    random_state = check_random_state(random_state)
-
     X = adata_comp.X
 
-    if chunked:
+    is_dask = isinstance(X, DaskArray)
+
+    random_state = check_random_state(random_state) if not is_dask else random_state
```

Comment:
why is that condition there?

Reply:
check_random_state returns a numpy RandomState object but dask expects an int

Reply:
```suggestion
    if not is_dask:
        random_state = check_random_state(random_state)
    elif not isinstance(random_state, int):
        msg = f'random_state needs to be an int, not a {type(random_state).__name__} when passing a dask array'
        raise TypeError(msg)
```